zebra/zebra_vrf.c

```diff
@@ -545,34 +518,20 @@ struct route_table *zebra_vrf_other_route_table(afi_t afi, u_int32_t table_id,
 						vrf_id_t vrf_id)
 {
 	struct zebra_vrf *zvrf;
-	rib_table_info_t *info;
-	struct route_table *table;
+	struct zebra_ns *zns;
 
 	zvrf = vrf_info_lookup(vrf_id);
 	if (!zvrf)
 		return NULL;
 
-	if (afi >= AFI_MAX)
-		return NULL;
+	zns = zvrf->zns;
 
-	if (table_id >= ZEBRA_KERNEL_TABLE_MAX)
```

Comment:
I see several other instances of `ZEBRA_KERNEL_TABLE_MAX` in zebra, especially in the handling of the `ip import-table (1-252)` command. We'll probably need to get rid of the `zebra_import_table_used` and `zebra_import_table_distance` arrays and reuse the `ns_tables` rb-tree to store that information. I'm ok if the intention is to do that work on a separate PR later.

Reply:
I actually intentionally did not do this work because I was not sure of how *bsd would react.  I know that since tables are only created internally in zebra when we receive information about it that leaving this alone for the moment was safe.

Additionally I was worried about breaking some subtle interaction of the import table command.